core/builtin.rbs

```diff
@@ -31,11 +31,11 @@ interface _Each[out A]
 end
 
 interface _Reader
-  def read: (?int length, ?string outbuf) -> String?
+  def read: (?int? length, ?string? outbuf) -> String?
```

Comment:
```suggestion
  def read: (?int? length, ?String outbuf) -> String?
```

I don't think making the type of the second parameter `string?` makes much sense, even though no runtime error raises. Is there a good reason to pass `nil` or non-string object with `#to_str` method?

Reply:
I'm thinking more about adoption/backwards-compatibility, given that there may be code somewhere passing some other object may or may not be nil, which type-checkers will complain about, and which just works. I agree that maybe we shouldn't allow nil from a type-checking perspective though. What do you think should be the best approach? `?string`?

Regarding the `string`, I'd leave it there. I've seen many buffer implementations on top of string objects, that make me believe users will pass smth other than strings there.

Reply:
Ok. It makes sense. `?string` seems the best type here to me.

Reply:
done